dev/integration_tests/flutter_gallery/test_driver/e2e_utils.dart

```diff
@@ -28,10 +28,19 @@ Future<void> watchPerformance(
     }
     return true;
   }());
+
+  // The engine could batch FrameTimings and send them only once per second.
+  // Delay for a sufficient time so either old FrameTimings are flushed and not
+  // interfering our measurements here, or new FrameTimings are all reported.
+  final Future<void> delayForFrameTimings =
+      Future<void>.delayed(const Duration(seconds: 2));
+
+  await delayForFrameTimings; // flush old FrameTimings
   final List<FrameTiming> frameTimings = <FrameTiming>[];
   final TimingsCallback watcher = frameTimings.addAll;
   binding.addTimingsCallback(watcher);
   await action();
+  await delayForFrameTimings; // make sure all FrameTimings are reported
```

Comment:
If the intention here is to wait for another 2 seconds, then I think you will need a fresh instance of `Future`.

Reply:
Good catch. Fixed.